sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/BlobDecryptionPolicy.java

```diff
@@ -163,30 +146,21 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
                 }
             });
         } else {
-            // Issue a get properties call to determine how much we will need to expand the range.
-            // Apply the request conditions from the download to the getProperties.
-            BlobRequestConditions rc = extractRequestConditionsFromRequest(requestHeaders);
-            return this.blobClient.getPropertiesWithResponse(rc).flatMap(response -> {
-                EncryptionData encryptionData = EncryptionData.getAndValidateEncryptionData(
-                    response.getValue().getMetadata().get(CryptographyConstants.ENCRYPTION_DATA_KEY),
-                    requiresEncryption);
-
-                // Apply the etag from the getProperties to the download to ensure consistency
-                String etag = response.getValue().getETag();
-                requestHeaders.set("ETag", etag);
-
-                // If there is no encryption data, then the blob is not encrypted. Pass the request through.
-                if (encryptionData == null) {
-                    return next.process();
-                }
+            // If it was a ranged request, we would have already called get properties and set encryption data.
+            // Since there is no encryption data, the request is not encrypted
+            if (context.getData(CryptographyConstants.ENCRYPTION_DATA_KEY).isEmpty()) {
+                return next.process();
+            }
+            EncryptionData encryptionData = EncryptionData.getAndValidateEncryptionData(
+                (String) context.getData(CryptographyConstants.ENCRYPTION_DATA_KEY).get(), requiresEncryption);
```

Comment:
should we store `EncryptionData` in the context instead of `String`?

Reply:
I'd de-serialize early and put strongly typed object in the context. In addition to error experience - it'll be handy if we need to add places that access it later.

Reply:
I updated this. FYI, though, this is only moved for ranged downloads. If it is not a ranged download (i.e. we intend to download the whole blob), we don't do a get properties and the encryption data is still deserialized in the policy.